### PR TITLE
Using Redux with Wechaty overview

### PR DESCRIPTION
**Pull request by shwetalsoni**
I have added **Using Redux with Wechaty** under Tutorials. The page is divided into some subsections as shown below:

Screenshot from 2021-06-30 13-47-03

The PR contains the updated submenu and **Overview** page. In later PR's we''ll include tutorials for respective subsections.

The overview page looks like this:

Screenshot from 2021-06-30 13-46-30
Screenshot from 2021-06-30 13-46-44

### CODE REVIEW

File: docusaurus/docs/using-redux-with-wechaty/overview.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+title: Wechaty Redux Overview
+sidebar_label: Overview
+---
+
+[![NPM Version](https://img.shields.io/npm/v/wechaty-redux?color=brightgreen)](https://www.npmjs.com/package/wechaty-redux)
+[![NPM](https://github.com/wechaty/wechaty-redux/workflows/NPM/badge.svg)](https://github.com/wechaty/wechaty-redux/actions?query=workflow%3ANPM)
+[![Ducksify Extension](https://img.shields.io/badge/Redux-Ducksify-yellowgreen)](https://github.com/huan/ducks#3-ducksify-extension-currying--api-interface)
+
+Using redux with wechaty wraps Wechaty with Redux Actions &amp; Reducers for Easy State Management.
```
Comment (proudofsimin) [inline docusaurus/docs/using-redux-with-wechaty/overview.md]: [W]echaty

File: docusaurus/docs/using-redux-with-wechaty/overview.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+[![Downloads](https://img.shields.io/npm/dm/wechaty-redux.svg?style=flat-square)](https://www.npmjs.com/package/wechaty-redux)
+[![TypeScript](https://img.shields.io/badge/%3C%2F%3E-TypeScript-blue.svg)](https://www.typescriptlang.org/)
+
+Redux is used for application state management and offers the following benefits:
+
+- Makes the state predictable
+
+- Easier to trace which action causes any change
+
+- Makes it easier to test, maintain and debug the code
```
Comment (proudofsimin) [inline docusaurus/docs/using-redux-with-wechaty/overview.md]: Use imperative sentence in the bulleted list, so please remove the s in makes.

File: docusaurus/docs/using-redux-with-wechaty/overview.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Vanilla Redux with Wechaty Redux Plugin
+
+Vanilla Redux means using plain Redux without any additional libraries like Ducks.
+
+The tutorial for **Vannila Redux with Wechaty Redux Plugin** is available [here](using-redux-with-wechaty/vannila-redux.md)
```
Comment (proudofsimin) [inline docusaurus/docs/using-redux-with-wechaty/overview.md]: You forgot the period at the end.

File: docusaurus/docs/using-redux-with-wechaty/overview.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+> Image Credit: [Alamy](https://www.alamy.com/cute-duck-and-little-ducks-over-white-background-colorful-design-vector-illustration-image185379753.html)
+
+[![Ducks Modular Proposal](https://img.shields.io/badge/Redux-Ducks%202015-yellow)](https://github.com/erikras/ducks-modular-redux)
+[![Re-Ducks Extended](https://img.shields.io/badge/Redux-Re--Ducks%202016-orange)](https://github.com/alexnm/re-ducks)
+[![Ducksify Extension](https://img.shields.io/badge/Redux-Ducksify%202020-yellowgreen)](https://github.com/huan/ducks#3-ducksify-extension-currying--ducksify-interface)
+
+Ducks offers a method of handling redux module packaging, installing, and running with your Redux store with middleware support.
+
+The tutorial for **Ducks Proposal Style for Wechaty Redux Plugin** is available [here](using-redux-with-wechaty/ducks-proposal.md)
```
Comment (proudofsimin) [inline docusaurus/docs/using-redux-with-wechaty/overview.md]: The period at the end.

**Review by proudofsimin — APPROVED** [review]
Just correct some minor mistakes and this is good :)

**Comment by shwetalsoni** [comment]
@proudofsimin I have made the suggested changes.

**Comment by shwetalsoni** [comment]
Tag #891

File: docusaurus/docs/using-redux-with-wechaty/overview.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+> Image Source: [Managing your React state with Redux](https://medium.com/the-web-tub/managing-your-react-state-with-redux-affab72de4b1)
+
+[![Downloads](https://img.shields.io/npm/dm/wechaty-redux.svg?style=flat-square)](https://www.npmjs.com/package/wechaty-redux)
+[![TypeScript](https://img.shields.io/badge/%3C%2F%3E-TypeScript-blue.svg)](https://www.typescriptlang.org/)
+
+Redux is used for application state management and offers the following benefits:
+
+- Make the state predictable
+
```
Comment (lijiarui) [inline docusaurus/docs/using-redux-with-wechaty/overview.md]: Pls remove blank lines here.

File: docusaurus/docs/using-redux-with-wechaty/overview.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+The tutorial for **Vannila Redux with Wechaty Redux Plugin** is available [here](using-redux-with-wechaty/vannila-redux.md).
+
+## Ducks Proposal Style for Wechaty Redux Plugin
+
+[![Ducksify Extension](https://img.shields.io/badge/Redux-Ducksify-yellowgreen)](https://github.com/huan/ducks#3-ducksify-extension-currying--api-interface)
+
+🦆🦆🦆Ducks is a Reducer Bundles Manager that implements the Redux Ducks Modular Proposal with great convenience.
+
+> Image Credit: [Alamy](https://www.alamy.com/cute-duck-and-little-ducks-over-white-background-colorful-design-vector-illustration-image185379753.html)
```
Comment (lijiarui) [inline docusaurus/docs/using-redux-with-wechaty/overview.md]: Where is the image? I think you may forget to upload the image here.

File: docusaurus/docs/using-redux-with-wechaty/overview.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+[![Ducksify Extension](https://img.shields.io/badge/Redux-Ducksify%202020-yellowgreen)](https://github.com/huan/ducks#3-ducksify-extension-currying--ducksify-interface)
+
+Ducks offers a method of handling redux module packaging, installing, and running with your Redux store with middleware support.
+
+The tutorial for **Ducks Proposal Style for Wechaty Redux Plugin** is available [here](using-redux-with-wechaty/ducks-proposal.md).
+
+## References
+
+- [NPM package of wechaty-redux](https://www.npmjs.com/package/wechaty-redux)
+
```
Comment (lijiarui) [inline docusaurus/docs/using-redux-with-wechaty/overview.md]: Pls remove blank line here.

**Review by lijiarui — CHANGES_REQUESTED** [review]
Pls change as I request.

File: docusaurus/docs/using-redux-with-wechaty/overview.md (hunk 6 above)
Comment (shwetalsoni) [reply docusaurus/docs/using-redux-with-wechaty/overview.md]: The image of **Ducks** is used above.  This is just the credit. Here I guess we have to provide only the link of the image used.

**Comment by shwetalsoni** [comment]
@lijiarui I have made the requested changes

File: docusaurus/docs/using-redux-with-wechaty/overview.md (hunk 6 above)
Comment (lijiarui) [reply docusaurus/docs/using-redux-with-wechaty/overview.md]: No, you didn't.

Here is the review of your PR
image

If you open this image https://www.alamy.com/cute-duck-and-little-ducks-over-white-background-colorful-design-vector-illustration-image185379753.html , you could find this image, but it didn't showed in the doc.

image

**Review by lijiarui — CHANGES_REQUESTED** [review]
Please fix image issue here.

**Comment by shwetalsoni** [comment]
> Please fix image issue here.

Actually, this image credit section was originally there on the page. So while updating it, I got confused with the image of ducks used above. I thought it was referring to that one. 

I think this image doesn't make sense here. Should I remove it?

**Comment by lijiarui** [comment]
> I think this image doesn't make sense here. Should I remove it?



> > Please fix image issue here.
> 
> Actually, this image credit section was originally there on the page. So while updating it, I got confused with the image of ducks used above. I thought it was referring to that one.
> 
> I think this image doesn't make sense here. Should I remove it?

I agree with you, remove the credit is a better solution.

**Comment by shwetalsoni** [comment]
@lijiarui I have removed the image credit.

**Review by lijiarui — APPROVED** [review]
LGTM

**Review by huan — APPROVED** [review]

**Comment by lijiarui** [comment]
@shwetalsoni Please resolve the conflict and then I will merge this PR.

**Pull request merged**